11_min_1_2.md

```diff
@@ -21,29 +21,47 @@ For most algorithms you also need objects.
 So you can design those after.
 All the best programmers are lazy.
 If they were not lazy, they would do work with their hands.
-They invented programming languages to be lazy.
-Imitate them.
+They invented programming languages to be lazy. Imitate them.
 
-## Combining binary counter and list pool
+## Overview
+
+We will call the function which finds the smallest and second smallest element `min_element1_2`.
+Note that I picked this algorithm not because it is of paramount importance
+for your future work.
+I pick this algorithm because it allows us to learn how to do decomposition
+and learn these components like `list_pool` and `binary_counter` along the way.
+
+Let me sketch the grand plan of the whole alogorithm.
```

Comment:
This information was completely missing before. This explains that we will use the linked list to store histories, and that it doesn't effect the binary_counter implementation.

Reply:
This is really good!